open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2165,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A flexible JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning. However it is recommended that servers
+        provide a plan-task-id, as well as a location to a manifest.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
```

Comment:
I think quite a few things are required, based on FileScanTaskParser.fromJson. 

And interestingly data-file is actually not required:

```
DataFile dataFile = null;
    if (jsonNode.has(DATA_FILE)) {
      dataFile = (DataFile) ContentFileParser.fromJson(jsonNode.get(DATA_FILE), spec);
    }
```

Although I am not sure why the parser decides to allow null for the data file. @stevenzwu could you help explain that?

Reply:
Added the other fields